compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -331,10 +314,5 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
           throw ex
       }
 
-    /** Transforms the rhs tree into a its default tree if it is in an `erased` val/def.
-    *  Performed to shrink the tree that is known to be erased later.
-    */
-    private def normalizeErasedRhs(rhs: Tree, sym: Symbol)(implicit ctx: Context) =
```

Comment:
What happens to erased right hand sides now? Are they kept? Why can't they be dropped here?

Reply:
The right-hand sides are kept as they were before. Implicit arguments to erased methods are removed early (as `(???: T)`). 

This logic was only droping nodes that where inlined in an erased rhs, the rest was unaffected.

Reply:
I tried to erase the rhs of erased val/def that are not inline. But YCheck complained on most erased tests. I will investigate further.

Reply:
I managed to implement it, now rhs of erased non inlined methods are removed in PostTyper